ibis/tests/all/test_geospatial.py

```diff
@@ -0,0 +1,241 @@
+
+
+@pytest.mark.parametrize(('arg', 'expected'), [
+    (polygon_0, [1.98] * 5),
+])
+@tu.skipifnot_backend(MapD)
+def test_area(backend, geo, arg, expected):
+    """Testing for geo spatial area operation."""
+    expr = arg(geo) if isfunction(arg) else arg
+    expr = geo[geo, expr.area().name('tmp')]
+    result = expr.execute()['tmp']
+    testing.assert_almost_equal(result, expected, decimal=2)
+
+
+@pytest.mark.parametrize(('arg', 'expected'), [
+    (lambda t: t['geo_linestring'], [1.41] * 5),
+])
+@tu.skipifnot_backend(MapD)
+def test_length(backend, geo, arg, expected):
+    """Testing for geo spatial length operation."""
+    expr = arg(geo) if isfunction(arg) else arg
+    result = expr.length().execute()
+    testing.assert_almost_equal(result, expected, decimal=2)
+
+
+@pytest.mark.parametrize(('arg', 'expected'), [
+    (lambda t: t['geo_polygon'], [5.66] * 5),
+])
+@tu.skipifnot_backend(MapD)
+def test_perimeter(backend, geo, arg, expected):
+    """Testing for geo spatial perimeter operation."""
+    expr = arg(geo) if isfunction(arg) else arg
+    result = expr.perimeter().execute()
+    testing.assert_almost_equal(result, expected, decimal=2)
+
+
+@pytest.mark.parametrize(('arg_left', 'arg_right', 'expected'), [
+    (lambda t: t['geo_linestring'], point_0, [1.41, 2.82, 4.24, 5.66, 7.08]),
+])
+@tu.skipifnot_backend(MapD)
+def test_max_distance(backend, geo, arg_left, arg_right, expected):
+    """Testing for geo spatial max-distance operation."""
+    expr = arg_left(geo) if isfunction(arg_left) else arg_left
+    result = expr.max_distance(
+        arg_right(geo) if isfunction(arg_right) else arg_right
+    ).execute()
+    testing.assert_almost_equal(result, expected, decimal=2)
+
+
+@pytest.mark.parametrize(('arg', 'expected'), [
+    (lambda t: t['geo_point'], [0, 1, 2, 3, 4]),
+])
+@tu.skipifnot_backend(MapD)
+def test_x(backend, geo, arg, expected):
+    """Testing for geo spatial x operation."""
+    expr = arg(geo) if isfunction(arg) else arg
+    result = expr.x().execute()
+    testing.assert_almost_equal(result, expected, decimal=2)
+
+
+@pytest.mark.parametrize(('arg', 'expected'), [
+    (lambda t: t['geo_point'], [0, 1, 2, 3, 4]),
+])
+@tu.skipifnot_backend(MapD)
+def test_y(backend, geo, arg, expected):
+    """Testing for geo spatial y operation."""
+    expr = arg(geo) if isfunction(arg) else arg
+    result = expr.y().execute()
+    testing.assert_almost_equal(result, expected, decimal=2)
+
+
+@pytest.mark.parametrize(('arg', 'expected'), [
+    (lambda t: t['geo_linestring'], [0, 1, 2, 3, 4]),
+])
+@tu.skipifnot_backend(MapD)
+def test_x_min(backend, geo, arg, expected):
+    """Testing for geo spatial x-min operation."""
+    expr = arg(geo) if isfunction(arg) else arg
+    result = expr.x_min().execute()
+    testing.assert_almost_equal(result, expected, decimal=2)
+
+
+@pytest.mark.parametrize(('arg', 'expected'), [
+    (lambda t: t['geo_linestring'], [1, 2, 3, 4, 5]),
```

Comment:
Most of the functions seem similar, You could merge the parametrizations:

```py
@pytest.mark.parametrize(('expr_fn', 'expected'), [
    (lambda t: t['geo_linestring'].x_max(), [1, 2, 3, 4, 5]),
    (lambda t: t['geo_linestring'].y_min(), [...]),
    (lambda t: t['geo_linestring'].y_max(), [...]),
    ...
)]
def test_geo_spatial_binops(backend, expr_fn, expected):
    """Testing for geo spatial x-max operation."""
    expr = expr_fn(geo)
    result = expr.execute()
    testing.assert_almost_equal(result, expected, decimal=2)
```

Reply:
Ok .. I split one test for function because I am planing to improve tests soon. but I can join some tests.

Reply:
I merged into 3 main test functions: unops, binops and get_point